src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -6488,6 +6488,18 @@ static SDL_GPUGraphicsPipeline *VULKAN_CreateGraphicsPipeline(
     return (SDL_GPUGraphicsPipeline *)graphicsPipeline;
 }
 
+static bool VULKAN_INTERNAL_IsValidShaderBytecode(
+    const Uint8 *code,
+    size_t codeSize)
+{
+    // SPIR-V bytecode has a 4 byte header containing 0x07230203 in the current
+    // platform's endianness.
+    if (codeSize < 4)
+        return false;
+    uint32_t magic = *(uint32_t*)code;
+    return magic == 0x07230203;
```

Comment:
I think the magic code can be LSB or MSB.
https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_magic_number

Reply:
Also, use `SDL_nemcmp` to avoid unaligned reads.

Reply:
SPIR-V can be either endianness within stored bytecode but I've found no evidence that drivers are able to automatically convert between the endiannesses internally. I think it's safer to be strict here and require the app to do the swizzle as they are the one loading the shader code from storage.

Reply:
I've changed this to accept both byte orders. It's still uncertain as to whether drivers can load both byte orders but for the purpose of testing for obviously-invalid shader code this is fine.